Websockets do check origin. If a server port is provided, the check will fail.

To avoid this problem, we add another default origin with a port (if headers to exist), thus
the default allowed origins will be:
  http(s)://<ip>
  http(s)://<ip>:<port>

--- a/src/ws/cockpitwebservice.c	2019-08-29 09:11:15.172902002 +0200
+++ b/src/ws/cockpitwebservice.c	2019-11-13 12:26:34.654122798 +0100
@@ -1307,7 +1307,7 @@
   const gchar **origins;
   gchar *allocated = NULL;
   gchar *origin = NULL;
-  gchar *defaults[2];
+  gchar *defaults[3];
   gboolean is_https;
   gchar *url;
 
@@ -1341,9 +1341,19 @@
   origins = cockpit_conf_strv ("WebService", "Origins", ' ');
   if (origins == NULL)
     {
+      const gchar *server_port = NULL;
+
       origin = g_strdup_printf ("%s://%s", is_https ? "https" : "http", host);
       defaults[0] = origin;
       defaults[1] = NULL;
+      defaults[2] = NULL;
+
+      if (headers)
+        {
+          server_port = g_hash_table_lookup (headers, "X-Forwarded-Port");
+          if (server_port)
+            defaults[1] = g_strdup_printf ("%s://%s:%s", is_https ? "https" : "http", host, server_port);
+        }
       origins = (const gchar **)defaults;
     }
 
